configure.ac

```diff
@@ -638,35 +636,10 @@ case $host in
      ;;
    *android*)
      dnl make sure android stays above linux for hosts like *linux-android*
-     LEVELDB_TARGET_FLAGS="-DOS_ANDROID"
+     TARGET_OS=android
```

Comment:
29283b67a82975ef195e43af42bae6cbe293356d: `TARGET_OS=android` is from 16110, which is complex and is not needed for leveldb backport I think
```suggestion
```